huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go

```diff
@@ -0,0 +1,223 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product DBSS
+// ---------------------------------------------------------------
+
+package dbss
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func DataSourceDbssFlavors() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceDbssFlavorsRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the name of the flavor.`,
+			},
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the availability zone which the flavor belongs to.`,
+			},
+			"level": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the level of the flavor.`,
+			},
+			"memory": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `Specifies the memory size(MB) in the flavor.`,
+			},
+			"vcpus": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `Specifies the number of CPUs.`,
+			},
+			"proxy": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `Specifies the maximum supported database instances.`,
+			},
+			"flavors": {
+				Type:        schema.TypeList,
+				Elem:        dbssFlavorsFlavorSchema(),
+				Computed:    true,
+				Description: `Indicates the list of DBSS flavors.`,
+			},
+		},
+	}
+}
+
+func dbssFlavorsFlavorSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the flavor.`,
+			},
+			"name": {
```

Comment:
as id and name are same, I think we can just keep `id`

Reply:
fixed